FloripaTrans/FTFindRoutesGoogleMapsViewController.m

```diff
@@ -0,0 +1,130 @@
+                                                                 zoom:14];
+    self.mapView.camera = camera;
+    
+    // Instantiate the route marker
+    self.routeMarker = [[GMSMarker alloc] init];
+    
+    // Change the back button so it has the title Back
+    UIBarButtonItem *item = [[UIBarButtonItem alloc] initWithTitle:@"Back" style:UIBarButtonItemStylePlain target:self action:@selector(goBack)];
+    self.navigationItem.hidesBackButton = YES;
+    self.navigationItem.leftBarButtonItem = item;
+    
+}
+
+- (void)goBack {
+    [self.navigationController popViewControllerAnimated:YES];
+}
+
+- (IBAction)okTouchUp:(id)sender {
+    if(!self.routeMarker.map) {
+        [self alertMessage:@"Choose a location" withTilte:@"Location Selection"];
+        return;
+    }
+    
+    [[UIApplication sharedApplication] setNetworkActivityIndicatorVisible:YES];
+    [self reverseGeocodeCoordinate:self.routeMarker.position];
+}
+
+
+/**
+ *  Method that check the content address based on a coordinate. If there's no error, the method calls a validation method to see if the address is a street or avenue.
+ *
+ *  @param coordinate coordinate
+ */
+- (void)reverseGeocodeCoordinate:(CLLocationCoordinate2D)coordinate {
+    GMSGeocoder *geocoder = [GMSGeocoder geocoder];
+    
+    [geocoder reverseGeocodeCoordinate:coordinate completionHandler:^(GMSReverseGeocodeResponse *response, NSError *error) {
+        
+        [[UIApplication sharedApplication] setNetworkActivityIndicatorVisible:NO];
+        
+        if(error) {
+            [self alertMessage:[error localizedDescription] withTilte:@"Map Error"];
+            
+            return;
+        }
+        
+        GMSAddress *address = [response firstResult];
+        if(address){
+           
+            [self checkForValidStreet:address.thoroughfare];
+         }
+    }];
+}
+
+/**
+ *  Check if the string is a street or an avenue. If it is valid, removes the AV. or R., call the delegate and pop the map view from the navigation stack. If it is not valid, it retrieves an erro message
+ *
+ *  @param string string which will be evaluated
+ */
+- (void) checkForValidStreet:(NSString *)string{
+    
+    NSError *error = nil;
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:@"(.*Av\\.\\s)|(.*R\\.\\s)" options:NSRegularExpressionCaseInsensitive error:&error];
+    
+    NSUInteger numberOfMatches = [regex numberOfMatchesInString:string
+                                                        options:0
+                                                          range:NSMakeRange(0, [string length])];
+    
+    if (numberOfMatches > 0) {
+        NSString *modifiedString = [regex stringByReplacingMatchesInString:string options:0 range:NSMakeRange(0, [string length]) withTemplate:@""];
+        
+        if(self.delegate && [self.delegate respondsToSelector:@selector(streetNameSelected:)]) {
```

Comment:
Oh, it definitely make sense. I opened issue #2 and I'm going to change it right away.